sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -99,10 +99,10 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
         throw QueryExecutionErrors.logicalPlanHaveOutputOfCharOrVarcharError(leaf)
 
       case u: UnresolvedNamespace =>
-        u.failAnalysis(s"Namespace not found: ${u.multipartIdentifier.quoted}")
+        throw new NoSuchNamespaceException(s"Namespace not found: ${u.multipartIdentifier.quoted}")
```

Comment:
One benefit of `u.failAnalysis` is that it can report the error position in SQL string. If we can't keep this feature with the new exception, I'd suggest we don't change the exception type.

Reply:
Should we close this PR or update `NoSuchNamespaceException` and `NoSuchTableException` to handle positions?

Btw, it looks like there is a bug in keeping the position. I will fix it separately.

Reply:
> ... update NoSuchNamespaceException and NoSuchTableException to handle positions?

If it's easy to do, let's go for it. Otherwise, we can close this PR as exception type seems not a big deal to me.